legend-sdlc-generation-file-maven-plugin/src/main/java/org/finos/legend/sdlc/generation/file/FileGenerationMojo.java

```diff
@@ -169,84 +198,121 @@ protected void serializeOutput(MutableMap<FileGenerationSpecification, List<Gene
         getLog().info(String.format("Done serializing %,d file generations' output to %s (%.9fs)", generationGenerationOutputMap.size(), this.outputDirectory, (System.nanoTime() - serializeStart) / 1_000_000_000.0));
     }
 
-    private void filterGenerationSpecsByIncludes(Map<String, GenerationSpecification> generationSpecsByPath) throws MojoExecutionException
+
+    protected void serializeArtifacts(MutableMap<ArtifactGenerationExtension, List<ArtifactGenerationResult>> results, Set<String> fileOutputPaths, PureModel pureModel) throws IOException, MojoExecutionException
+    {
+        long serializeStart = System.nanoTime();
+        int outputsCount = 0;
+        getLog().info("Start serializing artifact extension generations");
+        Path outputDirPath = this.outputDirectory.toPath();
+        Pattern pkgSepPattern = Pattern.compile("::", Pattern.LITERAL);
+        String replacement = Matcher.quoteReplacement(outputDirPath.getFileSystem().getSeparator());
+        for (Map.Entry<ArtifactGenerationExtension, List<ArtifactGenerationResult>> resultsPair : results.entrySet())
+        {
+            ArtifactGenerationExtension extension = resultsPair.getKey();
+            List<ArtifactGenerationResult> extensionResults = resultsPair.getValue();
+            for (ArtifactGenerationResult result: extensionResults)
+            {
+                org.finos.legend.pure.m3.coreinstance.meta.pure.metamodel.PackageableElement generator = result.getElement();
+                List<GenerationOutput> generatorOutputs = result.getResults();
+                for (GenerationOutput output: generatorOutputs)
+                {
+                    String elementFolder = Root_meta_pure_functions_meta_elementToPath_PackageableElement_1__String_1__String_1_(generator, FILE_PATH_DELIMETER, pureModel.getExecutionSupport());
```

Comment:
Use PackageableElement.getUserPathForPackageableElement instead to generate the folder. Also, use the filesystem separator. And can't you build this string once outside the generatorOutputs loop?